hydrocron/db/track_ingest.py

```diff
@@ -0,0 +1,163 @@
+
+    def _get_revision_start(self, collection_start_date):
+        """Locate the last most recent date that was queried in order to only
+        query on granules that have not seen before."""
+
+        # TODO - Query track-ingest table to determine max revision date
+        revision_start = None
+
+        if not revision_start:
+            revision_start = collection_start_date
+
+        return revision_start
+
+    def query_cmr(self):
+        """Query CMR to locate all granules for a specific time range.
+
+        Note: The use of "revision_date" should capture any granules that have
+        been reprocessed in the time range.
+        """
+
+        temporal_range = f"{self.revision_start.strftime('%Y-%m-%dT%H:%M:%SZ')},{self.revision_end.strftime('%Y-%m-%dT%H:%M:%SZ')}"
+        parameters = {
+            "short_name": self.collection_shortname,
+            "revision_date": temporal_range,
+            "page_size": self.PAGE_SIZE
+        }
+        logging.info("Search URL: %s", self.CMR_API)
+        logging.info("Search Parameters: %s", parameters)
+        cmr_response = requests.get(url=self.CMR_API, params=parameters, timeout=30)
+        self.cmr_granules = self._get_granule_ur_list(cmr_response.json())
+
+        hits = cmr_response.headers["CMR-Hits"]
+        total = len(self.cmr_granules.keys())
+
+        if "CMR-Search-After" in cmr_response.headers.keys():
+            search_after = cmr_response.headers["CMR-Search-After"]
+        else:
+            search_after = ""
+
+        headers = {}
+        while search_after:
+            logging.info("Searching for more results...%s out of %s", total, hits)
+            headers["CMR-Search-After"] = search_after
+            cmr_response = requests.get(url=self.CMR_API, headers=headers, params=parameters, timeout=30)
+            self.cmr_granules.update(self._get_granule_ur_list(cmr_response.json()))
+            total = len(self.cmr_granules.keys())
+            if "CMR-Search-After" in cmr_response.headers.keys():
+                search_after = cmr_response.headers["CMR-Search-After"]
+            else:
+                search_after = ""
+
+        for granule, revision_date in self.cmr_granules.items():
+            logging.debug("CMR granule located: %s  -  %s", granule, revision_date)
+        logging.info("Located %s granules in CMR.", len(self.cmr_granules.keys()))
+
+    @staticmethod
+    def _get_granule_ur_list(granules):
+        """Return dict of Granule URs and revision dates from CMR response JSON.
+
+        :param granules: Response JSON dictionary from CMR query
+        :type granules: dict
+
+        :rtype: dict
+        """
+
+        granule_dict = {}
+        for item in granules["items"]:
+            granule_ur = item["umm"]["GranuleUR"].replace("_swot", "")
+            checksum = 0
+            for file in item["umm"]["DataGranule"]["ArchiveAndDistributionInformation"]:
+                if f"{granule_ur}.zip" == file["Name"]:
```

Comment:
I think the generic way to find the data would involve looking in the `RelatedURLs` for `type == GET DATA` entries, extracting the filename from that entry and then looking in `ArchiveAndDistributionInformation` for files with a matching name to get the checksum for that file.

But that may be overkill since this is only working on the one collection. Do we agree that just looking for the `.zip` extension is good enough to identify the correct file within the list of granules? Is there a chance another .zip file would be added to these swot granules in the future?

Reply:
@torimcd - Do you know if another .zip file would be added to the SWOT granules? I am happy to modify this to be more precise and use the method that @frankinspace called out to determine the exact file name.

Reply:
@frankinspace @nikki-t, I don't think there's risk of another .zip file being added to the same granule in the future, at least for the SWOT collections. Looking for the only zip file in the granule should be a reliable way to find the correct data file here.